src/js/frontend/Dockerfile

```diff
@@ -0,0 +1,37 @@
+# syntax=docker/dockerfile:1.4
+
+# grapl-typescript-base
+################################################################################
+FROM node:18-bullseye-slim AS grapl-typescript-base
+SHELL ["/bin/bash", "-c"]
+
+RUN adduser \
+        --disabled-password \
+        --gecos '' \
+        --home /home/grapl \
+        --shell /bin/bash \
+        grapl
+# Uncomment to get the versions for pins
+#RUN dpkg-query -l && exit 5
+
+USER grapl
+WORKDIR /home/grapl
+ENTRYPOINT ["/bin/bash", "-o", "errexit", "-o", "nounset", "-c"]
+
+# integration-tests
+################################################################################
+FROM grapl-typescript-base AS integration-tests
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+# TODO: make sure yarn version matches package.json
+ENV YARN_VERSION 3.2.3
```

Comment:
This can be done similar to what we do with [Python](https://github.com/grapl-security/grapl/blob/main/build-support/python_version.sh) and [Rust](https://github.com/grapl-security/grapl/blob/main/build-support/rust_version.sh); the core logic would be:
```shell
jq --raw-output '.packageManager' src/js/frontend/package.json |
    awk 'BEGIN { FS="@" } { print $2 }'
```
We'd then pass it in as a build argument; see:
- https://github.com/grapl-security/grapl/blob/46f05bfbcfa8b73143bf90e1eaa198a018b6aa22/Makefile#L19-L21
- https://github.com/grapl-security/grapl/blob/46f05bfbcfa8b73143bf90e1eaa198a018b6aa22/docker-bake.hcl#L70-L76

This could be part of this PR, or a follow-up PR.